src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -2407,6 +2407,13 @@ TargetShardIntervalForFastPathQuery(Query *query, Const **partitionValueConst,
 		PruneShards(relationId, relationIndex, make_ands_implicit((Expr *) quals),
 					&queryPartitionValueConst);
 
+	if (PartitionMethod(relationId) != DISTRIBUTE_BY_NONE &&
+		(!queryPartitionValueConst || queryPartitionValueConst->constisnull))
+	{
+		/* we're not interested in NULL values */
```

Comment:
this comment seems unhelpful, what does a NULL value mean?

Reply:
updated now.